libs/pavex_session_sqlx/src/sqlite.rs

```diff
@@ -0,0 +1,909 @@
+
+        // Load and verify updates
+        let loaded = store.load(&session_id).await.unwrap().unwrap();
+
+        assert_eq!(
+            loaded.state.get("updated_field").unwrap(),
+            &serde_json::Value::String("new_value".to_string())
+        );
+        assert_eq!(
+            loaded.state.get("user_id").unwrap(),
+            &serde_json::Value::String("updated-user-456".to_string())
+        );
+
+        // Verify complex nested structure is preserved
+        let new_metadata = loaded.state.get("new_metadata").unwrap();
+        assert_eq!(new_metadata.get("last_action").unwrap(), "update_session");
+        assert_eq!(new_metadata.get("timestamp").unwrap(), 1640995260);
+
+        let deeply_nested = &new_metadata["complex_data"]["nested"]["deeply"];
+        assert_eq!(deeply_nested.as_array().unwrap().len(), 4);
+        assert_eq!(deeply_nested[0], "nested");
+        assert_eq!(deeply_nested[1], "array");
+        assert_eq!(deeply_nested[2], 123);
+        assert_eq!(deeply_nested[3], true);
+
+        // Verify TTL was updated
+        assert!(loaded.ttl.as_secs() > 7150);
+        assert!(loaded.ttl.as_secs() <= 7200);
+    }
+
+    #[tokio::test]
+    async fn test_ttl_expiry() {
+        let store = create_test_store().await;
+        let (session_id, state) = create_test_record(1);
+
+        let record = SessionRecordRef {
+            state: Cow::Borrowed(&state),
+            ttl: Duration::from_secs(1), // Very short TTL
+        };
+
+        // Create session with short TTL
+        store.create(&session_id, record).await.unwrap();
+
+        // Session should exist immediately
+        let loaded = store.load(&session_id).await.unwrap();
+        assert!(loaded.is_some());
+
+        // Wait for expiration
+        tokio::time::sleep(Duration::from_secs(2)).await;
+
+        // Session should be expired and not loadable
+        let expired = store.load(&session_id).await.unwrap();
+        assert!(expired.is_none());
+    }
+
+    #[tokio::test]
+    async fn test_update_ttl_roundtrip() {
+        let store = create_test_store().await;
+        let (session_id, state) = create_test_record(3600);
+
+        let record = SessionRecordRef {
+            state: Cow::Borrowed(&state),
+            ttl: Duration::from_secs(3600),
+        };
+
+        // Create session
+        store.create(&session_id, record).await.unwrap();
+
+        // Update TTL only
+        store
+            .update_ttl(&session_id, Duration::from_secs(7200))
+            .await
+            .unwrap();
+
+        // Verify TTL was updated but data preserved
+        let loaded = store.load(&session_id).await.unwrap().unwrap();
+        assert_eq!(
+            loaded.state.get("user_id").unwrap(),
+            &serde_json::Value::String("test-user-123".to_string())
+        );
+        assert!(loaded.ttl.as_secs() > 7150);
+        assert!(loaded.ttl.as_secs() <= 7200);
```

Comment:
I'd say it's enough to assert that it's greater than the original TTL.

Reply:
Changed